src/components/AnonymousChat/child/ChatItemMyTextV2.tsx

```diff
@@ -59,31 +81,37 @@ const ChatItemMyTextV2 = ({
         <IconChatCheckMark color={colors.silver} width={12} height={12} />
       </View>
     );
-  }, []);
+  }, [status]);
 
-  const contextMenuActions = [
-    {title: 'Reply', systemIcon: 'arrow.turn.up.left'},
-    {title: 'Copy Message', systemIcon: 'square.on.square'},
-    {title: 'Delete Message', systemIcon: 'trash', destructive: true}
+  const contextMenuActions: ContextMenuAction[] = [
+    {title: CONTEXT_MENU_REPLY, systemIcon: 'arrow.turn.up.left'}
   ];
 
+  if (messageType !== MESSAGE_TYPE_DELETED) {
+    contextMenuActions.push({title: CONTEXT_MENU_COPY, systemIcon: 'square.on.square'});
+    contextMenuActions.push({title: CONTEXT_MENU_DELETE, systemIcon: 'trash', destructive: true});
+  }
+
   const renderAvatar = React.useCallback(() => {
     if (!isShowUserInfo) return <View style={styles.avatar} />;
-    return <View style={styles.ml8}>{avatar}</View>;
-  }, []);
+    return <View style={styles.mlBuble}>{avatar}</View>;
+  }, [isShowUserInfo, avatar]);
 
-  const onSwipeToReply = (direction) => {
-    if (direction === 'right') return;
-    if (swipeableRef.current) swipeableRef.current?.close();
-    setReplyPreview({
-      username,
-      time,
-      message,
-      messageId: data?.id,
-      chatType,
-      messageType: 'regular'
-    });
-  };
+  const onSwipeToReply = React.useCallback(
+    (direction: 'left' | 'right') => {
+      if (direction === 'right') return;
+      if (swipeableRef.current) swipeableRef.current?.close();
+      setReplyPreview({
+        id: data?.id ?? data?.message?.id,
+        user: {username},
+        message,
+        message_type: messageType as MessageType,
+        updated_at: time,
+        chatType
+      });
+    },
+    [data, username, time, message, messageType, chatType]
+  );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `onSwipeToReply` function has been updated to include more data in the reply preview. This is a good improvement for providing more context in the reply preview. However, please ensure that all the fields (`id`, `user`, `message`, `message_type`, `updated_at`, `chatType`) are handled correctly in the reply preview component.



<!-- This is an auto-generated comment by OSS CodeRabbit -->